grails-app/controllers/au/org/ala/dataqualityfilter/QualityController.groovy

```diff
@@ -29,10 +29,11 @@ class QualityController {
             @ApiResponse(code = SC_OK, message = "OK", response = String, responseContainer = "Map")
     ])
     @ApiImplicitParams([
-            @ApiImplicitParam(name = "profileName", paramType = "query", required = false, value = "Profile name", dataType = 'string')
+            @ApiImplicitParam(name = "profileName", paramType = "query", required = false, value = "Profile name", dataType = 'string'),
+            @ApiImplicitParam(name = "userId", paramType = "query", required = false, value = "the userId used to get the default profile in case profile name is not provided", dataType = 'string')
     ])
-    def getEnabledFiltersByLabel(String profileName) {
-        render qualityService.getEnabledFiltersByLabel(profileName) as JSON
+    def getEnabledFiltersByLabel(String profileName, String userId) {
```

Comment:
This is an issue - You can't blindly trust the `userId` parameter here or on any other controller method because the caller could be anyone, not necessarily a trusted ALA system.  It needs like `@RequireApiKey` from `ala-ws-security-plugin` but instead of always applying only applies if the userId param is present.

Reply:
this will be done in next batch of changes